crates/sparrow-logical/src/typecheck.rs

```diff
@@ -31,6 +32,37 @@ pub(crate) fn typecheck(name: &str, args: &[ExprRef]) -> error_stack::Result<Typ
             };
             Ok(types)
         }
+        "record" => {
```

Comment:
Added special type-checking for records. This is part of why I'd like to make this a trait -- ideally, the type-checking logic is defined in the same place we specify how to map logical to physical.